### PR TITLE
Remove use of legacy codepath when establishing Postgres connections

### PR DESCRIPTION
**Pull request by gwynne**
Also tidies CI as usual.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,100 +27,120 @@ jobs:
   api-breakage:
     if: ${{ !(github.event.pull_request.draft || false) }}
     runs-on: ubuntu-latest
-    container: swift:5.8-jammy
+    container: swift:jammy
```
Comment (MahdiBM) [inline .github/workflows/test.yml:30]: Why this change? It's kind of inconsistent with all other CIs.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -133,43 +153,31 @@ jobs:
...
+          xcode-version: ${{ matrix.xcode-version }}
       - name: Install Postgres, setup DB and auth, and wait for server start
-        env:
-          POSTGRES_VERSION: ${{ matrix.dbimage }}
-          POSTGRES_AUTH_METHOD: ${{ matrix.dbauth }}
         run: |
-          export PATH="$(brew --prefix)/opt/${POSTGRES_VERSION}/bin:$PATH" PGDATA=/tmp/vapor-postgres-test
-          (brew unlink postgresql || true) && brew install "${POSTGRES_VERSION}" && brew link --force "${POSTGRES_VERSION}"
-          initdb --locale=C --auth-host "${POSTGRES_AUTH_METHOD}" -U "${POSTGRES_USER}" --pwfile=<(echo "${POSTGRES_PASSWORD}")
+          export PATH="$(brew --prefix)/opt/postgresql@14/bin:$PATH" PGDATA=/tmp/vapor-postgres-test
```
Comment (MahdiBM) [inline .github/workflows/test.yml:175]: why `@14`? Isn't the CI using postgres 15 now?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -27,100 +27,120 @@ jobs:
...
+    container: swift:jammy
     steps:
       - name: Check out code
         uses: actions/checkout@v3
         with: { 'fetch-depth': 0 }
       - name: Run API breakage check action
-        uses: vapor/ci/.github/actions/ci-swift-check-api-breakage@reusable-workflows
+        uses: vapor/ci/.github/actions/ci-swift-check-api-breakage@main
+
+  code-coverage:
```
Comment (MahdiBM) [inline .github/workflows/test.yml:38]: This one is just for my curiosity as you probably already considered it, but why not use the reusable CI to get rid of separate `code-coverage` and `gh-codeql` jobs?

**Review by MahdiBM — DISMISSED** [review]
The `@14`s seem suspicious 🤔

File: .github/workflows/test.yml (hunk 1 above)
Comment (gwynne) [reply .github/workflows/test.yml:30]: It's in line with https://github.com/vapor/ci/blob/main/.github/workflows/run-unit-tests.yml#L56 and https://github.com/vapor/ci/blob/main/.github/workflows/run-unit-tests.yml#L70 - these jobs don't care overmuch about which Swift version they use in particular, it should always just be the "latest stable" version, which is exactly what the non-versioned Docker tags provide.

File: .github/workflows/test.yml (hunk 2 above)
Comment (gwynne) [reply .github/workflows/test.yml:175]: The Linux unit tests cover Postgres versions 11, 13, and 15; we take the macOS tests as an excuse to cover 14 as well.

File: .github/workflows/test.yml (hunk 3 above)
Comment (gwynne) [reply .github/workflows/test.yml:38]: Same reason as the other non-SQLite Fluent repos don't - the reusable workflow doesn't yet support the services specifications these repos need and I haven't yet made the reusable workflow capable of invoking "only" the CodeQL/coverage/API breakage jobs without any tests.

**Review by MahdiBM — APPROVED** [review]
